syntax = "proto3";
import "empty.proto";

message UserInfo {
  int32 pk = 1;
  string email = 2;
  string profile_image = 3;
  string access_token = 4;
}

message LoginInfo {
  string email = 1;
  string password = 2;
}

service User {
  rpc Login(LoginInfo) returns (UserInfo) {};
  rpc Authenticate(Empty) returns (Empty) {}; // header 에 access-token 을 담아서 보낸다.
  rpc AuthenticateGetUserInfo(Empty) returns (UserInfo) {}; // header 에 access-token 을 담아서 보낸다.
}
